python/tests/algorithms/test_signals.py

```diff
@@ -0,0 +1,118 @@
+import sys
+import pytest
+import obspy
+import numpy as np
+
+import mspasspy.ccore as mspass
+from mspasspy.ccore import (Seismogram, TimeSeries, TimeSeriesEnsemble, SeismogramEnsemble)
+
+# module to test
+sys.path.append("../../mspasspy/util/")
+sys.path.append("../../mspasspy/algorithms/")
+sys.path.append("..")
+
+from helper import (get_live_seismogram,
+                    get_live_timeseries,
+                    get_live_timeseries_ensemble,
+                    get_live_seismogram_ensemble,
+                    get_stream,
+                    get_trace)
+
+from signals import (filter,
+                     detrend,
+                     interpolate,
+                     correlate,
+                     correlate_template,
+                     correlate_stream_template,
+                     correlation_detector,
+                     templates_max_similarity,
+                     xcorr_3c,
+                     xcorr_max,
+                     xcorr_pick_correction)
+
+def test_filter():
+    ts = get_live_timeseries()
+    seis = get_live_seismogram()
+    tse = get_live_timeseries_ensemble(3)
+    seis_e = get_live_seismogram_ensemble(3)
+    filter(ts, "bandpass", preserve_history=True, instance='0')
+    filter(seis, "bandpass", preserve_history=True, instance='0')
+    filter(tse, "bandpass", preserve_history=True, instance='0')
+    filter(seis_e, "bandpass")
+    filter(ts, "bandstop")
+    filter(ts, "lowpass")
+    filter(ts, "highpass")
+    filter(ts, "lowpass_cheby_2")
+    filter(ts, "lowpass_fir")
+    filter(ts, "remez_fir")
```

Comment:
This group of `filter` calls missed required frequency arguments e.g. `freqmin` and `freqmax` in bandpass.